text/0145-remove-unnecessary-allocator-usage.md

```diff
@@ -0,0 +1,1013 @@
+```
+
+##### Changes
+
+The function has already been using a runtime-allocated buffer to return its value. However, the result of the operation was returned as a host-allocated SCALE-encoded `Result`. It is changed to return a primitive indicating either the length written or an error.
+
+##### New prototype
+
+```wat
+(func $ext_offchain_http_response_read_body_version_2
+    (param $request_id i32) (param $buffer i64) (param $deadline i64) (result i64))
+```
+
+##### Arguments
+
+* `request_id` is an i32 integer indicating the ID of the started request, as returned by `ext_offchain_http_request_start`;
+* `buffer` is a pointer-size ([Definition 216](https://spec.polkadot.network/chap-host-api#defn-runtime-pointer-size)) to the buffer where the body is written;
+* `deadline` is a pointer-size ([Definition 216](https://spec.polkadot.network/chap-host-api#defn-runtime-pointer-size)) to the SCALE-encoded Option value ([Definition 200](https://spec.polkadot.network/id-cryptography-encoding#defn-option-type)) containing the UNIX timestamp ([Definition 191](https://spec.polkadot.network/id-cryptography-encoding#defn-unix-time)). Passing `None` blocks indefinitely.
+
+##### Result
+
+On success, the number of bytes written to the buffer is returned. A value of `0` means the entire response was consumed and no further calls to the function are needed for the provided request ID. On failure, a negative error code is returned, where `-1` denotes the deadline was reached, `-2` denotes that an I/O error occurred, and `-3` denotes that the request ID provided was invalid.
+
+#### ext_allocator_{malloc|free}
+
+##### Existing prototype
+
+```wat
+(func $ext_allocator_malloc_version_1 (param $size i32) (result i32))
+(func $ext_allocator_free_version_1 (param $ptr i32))
+```
+
+The functions are considered obsolete and must not be used in new code.
+
+#### ext_input_read
+
+##### Changes
+
+A new function providing means of passing input data from the host to the runtime. Previously, the host allocated a buffer and passed a pointer to it to the runtime. With the runtime allocator, it's not possible anymore, so the input data passing protocol changed (see "Other changes" section below). This function is required to support that change.
+
+##### New prototype
+
+```wat
+(func $ext_input_read_version_1
+    (param $buffer i64))
+```
+
+##### Arguments
+
+* `buffer` is a pointer-size ([Definition 216](https://spec.polkadot.network/chap-host-api#defn-runtime-pointer-size)) to the buffer where the input data will be written. If the buffer is not large enough to accommodate the input data, the function will panic.
+
+### Other changes
+
+Currently, all runtime entrypoints have the following identical Wasm function signatures:
+
+```wat
+(func $runtime_entrypoint (param $data i32) (param $len i32) (result i64))
+```
+
+After this RFC is implemented, such entrypoints are still supported, but considered deprecated. New entrypoints must have the following signature:
```

Comment:
> are still supported [..] New entrypoints must have the following signature

Why? This would imply that the host allocator will still be supported, which we ideally don't want to support anymore.

Also, the language here is inconsistent. In the next sentence you say "must", which sounds like they're not supported anymore?